spec/services/workflows/non_passported_workflow_spec.rb

```diff
@@ -10,44 +10,83 @@ module Workflows
 
     before do
       assessment.proceeding_type_codes.each do |ptc|
-        create :capital_eligibility, capital_summary: assessment.capital_summary, proceeding_type_code: ptc
-        create :gross_income_eligibility, gross_income_summary: assessment.gross_income_summary, proceeding_type_code: ptc,
-                                          upper_threshold: 20_000
+        create :capital_eligibility, upper_threshold: 20_000, capital_summary: assessment.capital_summary, proceeding_type_code: ptc
+        create :gross_income_eligibility, gross_income_summary: assessment.gross_income_summary, proceeding_type_code: ptc
         create :disposable_income_eligibility, disposable_income_summary: assessment.disposable_income_summary,
                                                lower_threshold: 500,
                                                proceeding_type_code: ptc
       end
     end
 
     describe ".call" do
-      subject(:workflow_call) { described_class.call(assessment) }
+      subject(:assessment_result) do
+        assessment.reload
+        described_class.call(assessment)
+        Assessors::MainAssessor.call(assessment)
+        assessment.assessment_result
+      end
+
+      before do
+        stub_request(:get, "https://www.gov.uk/bank-holidays.json")
+          .to_return(body: file_fixture("bank-holidays.json").read)
+
+        assessment.proceeding_type_codes.each do |ptc|
+          create(:assessment_eligibility, assessment:, proceeding_type_code: ptc)
+        end
+      end
 
       context "when self_employed" do
-        let(:applicant) { create :applicant, self_employed: true }
+        let(:applicant) { build :applicant, self_employed: true }
 
         it "calls the self-employed workflow" do
           expect(SelfEmployedWorkflow).to receive(:call).with(assessment)
-          workflow_call
+          described_class.call(assessment)
         end
       end
 
-      context "without capital distractions" do
-        let(:applicant) { create :applicant, :over_pensionable_age, self_employed: false, employed: }
-        let(:assessment_result) do
-          assessment.reload
-          workflow_call
-          Assessors::MainAssessor.call(assessment)
-          assessment.assessment_result
+      context "with capital" do
+        before do
+          create(:property, :additional_property, capital_summary: assessment.capital_summary,
+                                                  value: 170_000, outstanding_mortgage: 100_000, percentage_owned: 100)
         end
 
-        before do
-          stub_request(:get, "https://www.gov.uk/bank-holidays.json")
-            .to_return(body: file_fixture("bank-holidays.json").read)
+        context "without partner" do
```

Comment:
I'm increasingly of the opinion that this stuff should _also_ be tested in BA-readable cucumber files. But thanks to the piecemeal way we've been putting this stuff together it's not really been possible til now. Do you think we should have a separate ticket to add cucumber tests?